src/o365/todo/commands/list/list-remove.ts

```diff
@@ -0,0 +1,166 @@
+import commands from '../../commands';
+import request from '../../../../request';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate, CommandError
+} from '../../../../Command';
+import GraphCommand from '../../../base/GraphCommand';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id?: string;
+  name?: string;
+  confirm?: boolean;
+}
+
+class TodoListRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.LIST_REMOVE}`;
+  }
+
+  public get description(): string {
+    return 'Removes a Microsoft To Do task list';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+
+    const getListId = () => {
+      if (args.options.name) {
+        // Search list by its name
+        const requestOptions: any = {
+          url: `${this.resource}/beta/me/todo/lists?$filter=displayName eq '${escape(args.options.name)}'`,
+          headers: {
+            accept: "application/json;odata.metadata=none"
+          },
+          json: true
+        };
+        return request.get(requestOptions)
+          .then((response: any) => response.value && response.value.length === 1 ? response.value[0].id : null);
+      }
+
+      return Promise.resolve(args.options.id);
+    };
+
+    const removeList = () => {
+      getListId().then(listId => {
+        if (!listId) {
+          cb(new CommandError(`The list ${args.options.name} cannot be found`));
+          return;
+        }
```

Comment:
This was a comment that was missed 😬

If we supply the `id` of a task list that doesn't exist, the Graph API doesn't throw an error, instead it returns a `204` success response, thus resulting in a false positive.

We should check the validity of the `id` that is passed and throw an error in the same way we do it for when you pass the `name` of a task list that doesn't exist.

Reply:
Okay, something that we will have to take care of...
However some observations with the beta API:
- with any Id to `https://graph.microsoft.com/beta/me/todo/lists/?$filter=id eq 'AAMkAGI3NDhlZmQzLWQxYjAtNGJjNy04NmYwLWQ0M2IzZTNlMDUwNAAuAAAAAACQ1l2jfH6VSZraktP8Z7auAQCbV93BagWITZhL3J6BMqhjAAD9pHIjAAA='`
![image](https://user-images.githubusercontent.com/8345556/82766791-1e246600-9e22-11ea-8d33-f2adf174eb13.png)
- with a quite different Id `https://graph.microsoft.com/beta/me/todo/lists/AADDDGI3NDhlZmQzLWQxYjAtNGJjNy04NmYwLWQ0M2IzZTNlMDUwNAAuAADDDDCQ1l2jfH6VSZraktP8Z7auAQCbV93BagWITZhL3J6BMqhjAAD9pHIjAAA=`
![image](https://user-images.githubusercontent.com/8345556/82766827-896e3800-9e22-11ea-9473-d240b784c6e7.png)
- **But worst !** with a slightly mispelled Id `https://graph.microsoft.com/beta/me/todo/lists/AAMkAGI3NDhlZmQzLWQxYjAtNGJjNy04NmYwLWQ0M2IzZTNlMDUwNAAuAAAAAACQ1l2jfH6VSZraktP8Z7auAQCbV93BagWITZhL3J6BMqhjAAD9pHIjDDD=` (Only the 3 last characters mispelled)
![image](https://user-images.githubusercontent.com/8345556/82766856-d520e180-9e22-11ea-82f0-4715a51176ef.png)

Bottom line is, the APIs are clearly not yet reliable for IDs....

Reply:
Actually... Identifying by the display name is more reliable... Maybe we should get rid of the Id option altogether for now ?

Reply:
Interesting observations, thank you for the research and context on this.

The third example is worrying if you get a different task list returned that doesn't match the Id supplied in the request, we should definitely relay this back to the Graph team.

My test just passed the same Id of the list I had just deleted, essentially running the command twice and expecting an error on the second try.

Based on your observations, we could call `GET /beta/me/todo/lists/<listId>` and then confirm that the success response does not match with the Id supplied and throw an error, however this is not ideal. Thoughts?

Reply:
It looks pretty straightforward! That should work ! Let’s give it a try!

Reply:
@pnp/office-365-cli-maintainers are you happy with this approach?

Reply:
I still don't get where the Id come from when you mispell it and it returns something that exists with neither the original Id nor the Id you specified XD ... But anyway, with checking if the argument Id and the return Id are the same, we should be okay, I tested it and it works... I am adapting the .spec file

Reply:
@garrytrinder as you mentioned, with the Id of the list you've just deleted, you get some expected 404 which is handled by the command error handler and prints out _Item not found_, if you misspell the Id, you get this strange behavior. but then,  it goes through this code I just changed
```TypeScript
    if (!listId) {
          if (args.options.name) {
            cb(new CommandError(`A task list with name ${args.options.name} cannot be found`));
          } else if (args.options.id) {
            cb(new CommandError(`A task list with Id ${args.options.id} cannot be found`));
          }
          return;
        }
```

To be consistent to the user, should we change these both messages to _Item not found_ as well ?

Reply:
Thanks you for your efforts on this @ypcode , however I think we should put a pause on this for the time being rather than trying to code around the issues and avoid unnecessary work.

We reached out to the Graph team about these endpoints and although they are available in the Graph, they appear to be in an alpha state, hence there are no docs.

We expect that there will be changes made before it reaches beta status, therefore we should wait for comms around this release and docs are added to support these endpoints.